src/python/pants/backend/native/tasks/native_external_library_fetch.py

```diff
@@ -69,36 +77,28 @@ class NativeExternalLibraryFiles(datatype([
     # TODO: we shouldn't have any `lib_names` if `lib_dir` is not set!
     'lib_dir',
     ('lib_names', tuple),
+    ('static_archive_paths', tuple),
```

Comment:
Any particular reason things don't just work? IIRC, simply passing `-l<libname>` will trigger a search of supplied search dirs, right? Passing the direct path to the static archive is better than two separate options for the search dir and the library name, but since we are already doing that, why add more complexity?

Reply:
It didn't work when I tried it! I don't know why, or if this is accepted in some places but not others. I'm pretty sure this is a valid command line invocation regardless, but I definitely tried `-L`/`-l` and it didn't work until I did this.

Reply:
I 100% agree that it is more complexity, and it would probably be great to leave a TODO here.

Reply:
Yeah ok so it's *supposed* to work with `-L`/`-l`, as you describe, but it may be that clang doesn't understand that. The option `-static` is supposed to use static versions of libraries whenever possible, but all my attempts to make that work (which didn't use even more complex hacks) either errored at link time or at runtime when the ctypes library is loaded. This should be in a comment in the code.

Reply:
`IrrXML` only provides a static lib, no dynamic version, or I'd suspect the linker just got confused.

Reply:
Thanks for clarifying, sounds good.